### ISSUE TITLE
Does not work with adguard

### ISSUE THREAD

**Issue by Nirgranth** [safing/portmaster#431]
### What worked?
Everything worked when either adguard or postmaster were off 

### What did not work?
DNS errors on the entire internet when both adguard and postmaster are on

### Additional information

**Debug-Info**: https://support.safing.io/privatebin/?9b5ea6910306cb75#5hk4ck9xdAkdXNNXdzddvp5gBt3qPTMqPpDDm1yGxVto

**Comment by Nirgranth** [comment]
it just does not accept them as it is as incoming connection block rule in Privacy Filter Module, Adding a exception for them works, but ips might change on restart, (i will updated after checking),

![Screenshot 2021-11-11 100600](https://user-images.githubusercontent.com/48947296/141238336-c1984bcb-8d92-472b-abc9-151e4d6cfba1.png)
![png_20211014_112632_0000](https://user-images.githubusercontent.com/48947296/141238338-908e13fa-0965-42a5-af80-85b8017ebf86.png)
![Screenshot 2021-11-11 100330](https://user-images.githubusercontent.com/48947296/141238339-f529462b-3545-4cfa-a9e5-c5e60c946048.png)
![Screenshot 2021-11-11 100530](https://user-images.githubusercontent.com/48947296/141238341-06af253a-368d-4b40-a342-87093eae7b3e.png)
![Screenshot 2021-11-11 100542](https://user-images.githubusercontent.com/48947296/141238342-1d318e11-e537-4eb1-a796-f1a889057760.png)

**Comment by Nirgranth** [comment]
it still blocks some things from system
![Screenshot 2021-11-11 103324](https://user-images.githubusercontent.com/48947296/141240530-737ca031-8f5e-4338-a949-f251531fc202.png)
![Screenshot 2021-11-11 103344](https://user-images.githubusercontent.com/48947296/141240536-5e4e375f-c386-43f5-9dfc-8157c74b7391.png)

**Comment by dhaavi** [comment]
Thanks for the screenshots!

I'm not sure how exactly you are running adguard. Are you running it on your router or on your device directly?
If you are worried about IPs changing, you can also use an "Allow `LAN`" rule instead.

**Comment by Nirgranth** [comment]
Adguard is currently running on the device. i am not sure switching it to a pi hole would help, 
Thanks Allow Lan works fine.
By default it works because it allows incoming connections on ports 5535 and 1900, and adguard seems to send them on random ports. 
Might have to configure adguard to send on those ports. (will check if thats possible and update)

![Screenshot 2021-11-11 180128](https://user-images.githubusercontent.com/48947296/141298902-d0b56a50-c089-478c-aecd-de59c9460bb6.png)

**Comment by Nirgranth** [comment]
i am pretty sure adguard filter dns requests before portmaster, then they get blocked by portmaster.

**Comment by davegson** [comment]
as I understand AdGuard is compatible when adding the `Allow LAN` rule - added this to the reports and hence closing this issue.

If I misunderstood please let me know

**Comment by celphogeek** [comment]
> Thanks for the screenshots!
> 
> I'm not sure how exactly you are running adguard. Are you running it on your router or on your device directly? If you are worried about IPs changing, you can also use an "Allow `LAN`" rule instead.

Where can I do this in portmaster?

**Comment by Nirgranth** [comment]
![image](https://user-images.githubusercontent.com/48947296/151266720-89a39754-3b57-4e39-8342-cecd2c0eb057.png)

**Comment by dhaavi** [comment]
Hey @celphogeek, thanks for asking this question and thanks @Nirgranth for the screenshot.

Please keep in mind that we don't have closed issues on our radar - we literally don't see them with how we manage issues. In case you an issue needs more attention, please reopen it, or create a new issue.

**Comment by itsnotsaved** [comment]
@Nirgranth does that adguard desktop?

**Comment by nirgranth** [comment]
use 

![Image](https://github.com/user-attachments/assets/edfa6bbc-1e2e-4c20-a8fa-a11ed0730ee4)

or turn off

![Image](https://github.com/user-attachments/assets/99025347-9a91-4954-ab63-3e03c695e8b7)

**Comment by alphanightshift** [comment]
> use
> 
> ![Image](https://github.com/user-attachments/assets/edfa6bbc-1e2e-4c20-a8fa-a11ed0730ee4)
> 
> or turn off
> 
> ![Image](https://github.com/user-attachments/assets/99025347-9a91-4954-ab63-3e03c695e8b7)

Even if you set up to allow other DNS to run on AdGuard for Windows, there might be a chance a compatibility issue arise in Portmaster, as the icon/notification turns red (V2 has a bug that does not correctly show the status Secure/Insecure/Warning). All versions are affected.
The best way, for me at least, is to set up the DNS Protection in AdGuard to 127.0.0.1, port 853 or 53 your choice.
Downsides are your internet connection may be slightly delayed, and some VPNs, particularly ProtonVPN (I tested on version 4.2.1), may not work with this setup, even if you set Secure DNS Bypassing to off or set DNS to 127.0.0.1.

**Comment by oruphonics1** [comment]
> use
> 
> ![Image](https://github.com/user-attachments/assets/edfa6bbc-1e2e-4c20-a8fa-a11ed0730ee4)
> 
> or turn off
> 
> ![Image](https://github.com/user-attachments/assets/99025347-9a91-4954-ab63-3e03c695e8b7)

After a lot of tries, I think this should be the way to let AdGuard (version 7.22.3 latest) fully compatible with Portmaster while reducing most of the annoying drawbacks I know.
1. Uninstall Portmaster, Adguard.
2. Restart the computer, then install Portmaster first (v1 latest is recommended as it is more stable to VPN and UI is native, less bugs).
3. Restart, then install Adguard.
4. Restart, then let both run. Let Portmaster capture Adguard Svc traffic then shut Adguard down.
From this point on, there are two ways to configure both of the apps. The former one in my experience is more stable and less prone to conflicts.
First choice:
- Configure Portmaster like in the quote with a few changes:
Outgoing: Only use the last three allow rules.
Incoming: Add another Block rule *.
- Configure Adguard:
1. Settings -> General Settings -> Advanced Settings -> Exclude apps from filtering. At there, whitelist the path where portmaster-core.exe is in.
2. Settings -> DNS Protection -> create a custom DNS with DNS upstream 127.0.0.1. Use that DNS.
3. Settings -> Network -> uncheck Use WFP network driver. This is to avoid HTTPS filtering from Adguard not working due to using the same WPF driver as PM.
Second way:
- Configure Portmaster the same in the first choice, but also whitelist your preferred DNS server you intend to setup in Adguard in Outgoing rules. Set it at the top.
- Configure Adguard.
Do step 1 as in first choice. In step 2, choose or create custom DNS server you like. Then do 3 as in first choice.
5. Restart your computer.

**Comment by nirgranth** [comment]
@oruphonics1 i went through a lot when it didnt work, glad you figured it out

**Comment by oruphonics1** [comment]
> [@oruphonics1](https://github.com/oruphonics1) i went through a lot when it didnt work, glad you figured it out

tbh even with this setup this is not really reliable. Certain VPNs (Proton, for example, in my experience), might crash Adguard all of a sudden.
TDI driver from Adguard is also on the way to be depreciated, and it also decreases your network speed noticeably. And if you also use Chrome (you better not) or some Chromium-based browser, TDI driver may not even work depending on your setup.
Luckily their new driver, SockFilter, is on the way to be implemented (finally) on version 7.x and later on 8.x (gonna take a long time there), so I hope at that point Adguard and Portmaster will play nicer with each other.
